models/fivetran_platform__connector_daily_events.sql

```diff
@@ -124,20 +127,22 @@ join_event_history as (
 
     from
     spine left join connector_event_history
-        on cast(spine.date_day as date) = connector_event_history.date_day
+        on spine.date_day = connector_event_history.date_day
 
-    group by 1,2,3,4,5,6
+    group by spine.date_day, connector_name, connector_id, connector_type, destination_name, destination_id
 ),
 
 final as (
 
     select *
     from join_event_history
 
-    where cast(date_day as timestamp) <= {{ dbt.current_timestamp_backcompat() }}
+    where date_day <= cast({{ dbt.current_timestamp_backcompat() if target.type != 'sqlserver' else dbt.current_timestamp() }} as date)
```

Comment:
This was ultimately because `current_timestamp_backcompat` just did not work with sqlserver

based on this [comment](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/adapters/timestamps.sql#L20-L28) I felt `current_timestamp` could be appropriate, as this package on SQLserver is a "new project" kinda and has no past to be backcompatible with

Reply:
Interesting and that's a great observation. With that, I am comfortable taking the approach you have applied here. Thanks!